### DISCUSSION TITLE
Can't get Ghostscript working with ImageMagick on CentOS 8

### DISCUSSION THREAD

**Discussion by r14smit** [ImageMagick/ImageMagick#2771] [Help]
I like to convert jpg and pdf files using Ghostscript (gs) as the delegate. For some reason, I can't get this working, while I have it working on CentOS 7.

I had GhostScript 9.53.3 installed before I installed the latest version of ImageMagick 7.0.10-34.

testing the convert :
convert -debug all ADV_FP.pdf advert.jpg

results in:
convert: no decode delegate for this image format `' @ error/constitute.c/ReadImage/560.
convert: no images defined `advert.jpg' @ error/convert.c/ConvertImageCommand/3273

Any suggestion to solve this would be appreciated

**Comment by dlemstra** [comment]
It might be possible that in CentOS 8 they disabled Ghostscript by default in the policy of ImageMagick. And did you install gslib or did you install the executable?

**Reply by r14smit** [reply]
I installed the ghostpdl-9.53.3, so no executable. Also added the following lines to the policy:

```
  <policy domain="module" rights="read|write" pattern="{PS,PDF,XPS}" />
  <policy domain="delegate" rights="read|write" pattern="gs" />

```
also added the path of gs in the delegates.xml like:
<delegate decode="ps" encode="eps" mode="bi" command="&quot;**/usr/local/bin/**gs&quot; -sstdout=%%stderr -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 &quot;-sDEVICE=eps2write&quot; &quot;-sOutputFile=%o&quot; &quot;-f%i&quot;"/>

did it for all 'gs' declarations.

config.log mentions following on Ghostscript:

configure:31219: checking for Ghostscript
configure:31221: result:
configure:31226: checking ghostscript/iapi.h usability
configure:31226: gcc -c  -fopenmp -g -O2 -Wall -mtune=ivybridge -fexceptions -pthread -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16   -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16  conftest.c >&5
conftest.c:168:10: fatal error: ghostscript/iapi.h: No such file or directory
 #include <ghostscript/iapi.h>
          ^~~~~~~~~~~~~~~~~~~~
ac_cv_header_ghostscript_iapi_h=no
ac_cv_header_ghostscript_ierrors_h=no

**Reply by dlemstra** [reply]
I think you will need to install `libgs-devel` to link with Ghostscript instead.
